pkg/webhook/work/mutating.go

```diff
@@ -106,4 +111,54 @@ func removeIrrelevantField(workload *unstructured.Unstructured) {
 			unstructured.RemoveNestedField(workload.Object, "spec", "clusterIP")
 		}
 	}
+
+	if workload.GetKind() == util.JobKind {
+		job := &batchv1.Job{}
+		err := runtime.DefaultUnstructuredConverter.FromUnstructured(workload.UnstructuredContent(), job)
+		if err != nil {
+			return err
+		}
+
+		if job.Spec.ManualSelector == nil || !*job.Spec.ManualSelector {
+			return removeGenerateSelectorOfJob(workload)
+		}
+	}
+
+	return nil
+}
+
+func removeGenerateSelectorOfJob(workload *unstructured.Unstructured) error {
```

Comment:
The `removeGenerateSelectorOfJob` is anti-operation of [generateSelector](https://github.com/kubernetes/kubernetes/blob/5a732dcfe1d4ec0e8ee2871b106605b7f8a69b98/pkg/registry/batch/job/strategy.go#L215:6), we can't track the change when updating kubernetes, (the behavior might be different), so an E2E test is necessary.